Update SBT and plugins to latest versions

retronym opened the pull request.

retronym committed:
Update SBT and plugins to latest versions

retronym commented:
@SethTisue What's the best way to do a re-run of the 2.13.6 packages with this PR to let us compare the artifacts?

retronym commented:
Last sbt-native-packager upgrade issue #166, PR #180

retronym committed:
Diagnostic for failed Wix compilation

SethTisue commented:
> What's the best way to do a re-run of the 2.13.6 packages with this PR to let us compare the artifacts?

It should be safe to run the usual release steps as far as the creation of the staging repos, but whether there is something along the way that will make it fail when Maven Central already has artifacts with the same version number, I simply don't know. Perhaps you'll need to make up a version number 2.13.6-2 or something and accept that the artifact-comparing will be harder.

retronym commented:
I'm going for local runs, using VMs to create and test installs of the .msi and .dep bundles.

retronym commented:
Looking good so far. Zip are identical according to `zipcmp` (which ignores timestamps).

Ideally we'd set the all timestamps deterministically to make the build repeatable, but that's one for another day.

```
$ git --git-dir ~/code/scala-dist-baseline/.git show --stat
commit 9b9c8955594b38821aa823846f68f483f7044c6a (HEAD -> 2.13.x, origin/HEAD, origin/2.13.x)
Merge: 632d27b 288038e
Author: Seth Tisue <seth@tisue.net>
Date:   Tue Jun 15 21:33:09 2021 -0700

    Merge pull request #217 from retronym/merge/2.12.x-to-2.13.x-20210616

 .travis.yml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
ubuntu@ubuntu1:/tmp/baseline$ git --git-dir ~/code/scala-dist/.git show --stat
commit 83451fd8df77bad86a1480fee0e1ae58dec3eae1 (HEAD -> topic/modernize-sbt, scala-dist-baseline)
Author: Jason Zaugg <jzaugg@gmail.com>
Date:   Tue Jun 15 16:01:30 2021 +1000

    Diagnostic for failed Wix compilation

 appveyor.yml      | 1 +
 project/Wix.scala | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

$ zipcmp -t -v /home/ubuntu/code/scala-dist-baseline/target/universal/scala-2.13.6.zip /home/ubuntu/code/scala-dist/target/universal/scala-2.13.6.zip

$ zipcmp /home/ubuntu/code/scala-dist-baseline/target/universal-docs/scala-docs-2.13.6.zip /home/ubuntu/code/scala-dist/target/universal-docs/scala-docs-2.13.6.zip
```

.rpm and .deb are extremely close in size, but I haven't analyzed the differences yet.

```
$ ls -la /home/ubuntu/code/scala-dist-baseline/target/scala-2.13.6.rpm /home/ubuntu/code/scala-dist/target/scala-2.13.6.rpm
-rw-rw-r-- 1 ubuntu ubuntu 138224796 Jun 16 18:02 /home/ubuntu/code/scala-dist-baseline/target/scala-2.13.6.rpm
-rw-rw-r-- 1 ubuntu ubuntu 138258506 Jun 16 16:28 /home/ubuntu/code/scala-dist/target/scala-2.13.6.rpm

$ ls -la /home/ubuntu/code/scala-dist-baseline/target/scala_2.13.6-400_all.deb /home/ubuntu/code/scala-dist/target/scala_2.13.6-400_all.deb
-rw-rw-r-- 1 ubuntu ubuntu 678368332 Jun 16 18:02 /home/ubuntu/code/scala-dist-baseline/target/scala_2.13.6-400_all.deb
-rw-rw-r-- 1 ubuntu ubuntu 678361638 Jun 16 16:28 /home/ubuntu/code/scala-dist/target/scala_2.13.6-400_all.deb
```

retronym committed:
isSnapshot may be false for manually overriden versions

When using `sbt -Dscala.version=2.X.Y` to test out packages,
newer version of sbt-git have been setting `isSnapshot` to false
if there isn't a tag associated with the current commit of
scala/scala-dist. This makes it hard to test out changes in
this repo as we get diffs like:

```
$ rpm -qip /home/ubuntu/code/scala-dist-baseline/target/scala-2.13.6.rpm
Name        : scala
Version     : 2.13.6
Release     : 1
Architecture: noarch
Install Date: (not installed)
Group       : Development/Languages
Size        : 649267584
License     : BSD
Signature   : (none)
Source RPM  : scala-2.13.6-1.src.rpm
```

```
$ rpm -qip  /home/ubuntu/code/scala-dist/target/scala-2.13.6.rpm
Name        : scala
Version     : 2.13.6
Release     : SNAPSHOT
Architecture: noarch
Install Date: (not installed)
Group       : Development/Languages
Size        : 649267584
License     : BSD
Signature   : (none)
Source RPM  : scala-2.13.6-SNAPSHOT.src.rpm
...
```

retronym commented:
The difference was a metadata difference. sbt-git is now setting the `isSnapshot` build setting to `true` when overriding the project with `-Dproject.version=2.13.6`. This propagates into the RPM metadata. More details in last commit message that works around this behaviour change.

retronym commented:
This one is ready to go from my perspective.

dwijnand commented on admin/build.ps1:
```diff
@@ -49,6 +49,6 @@ if ($env:APPVEYOR_FORCED_BUILD -eq 'true') {
   clearIvyCache
```
Also change 2 lines up to 2.13.6?

retronym committed:
Bump default version to 2.13.6

dwijnand commented on project/ScalaDist.scala:
```diff
@@ -57,31 +59,32 @@ object ScalaDist {
+    "org.scala-lang" % "scala-dist" % v % s"${ScalaDistConfigName}; ${ScalaDistConfigName}->${config}"
 
-  // map module to the corresponding file mappings (unzipping scala-dist in the process)
+  // map module to the corresponding file mappings (unzipping scala-the / dist process)
```
lol

retronym replied:
oops, fixed

retronym committed:
Undo incorrect or temp changes
